File: IE 211 Lab/src/Lab3/Difference.java
package Lab3;

public class Difference {
	
	public Difference() {}
	
	public void printDifferentElements(int[] array1, int[] array2) 
	{
		System.out.println("The elements of A that are not in B are ");
		
		for(int i = 0; i < array1.length; i++)
		{
			boolean isElement = false;
			for(int j = 0; j < array2.length; j++)
			{
				if(array1[i] == array2[j])
				{
					isElement = true;		
					break;
				}		
			}
			
			if(isElement == false)
				System.out.print(array1[i] + " ");
		}		
	}

}

File: IE211 Fall22/src/Lab4/Difference.py
class Difference:

    def __printDifferentElements_v1(self, array1, array2):
        print("default version")

        for i in range(len(array1)):
            isElement = False
            for j in range(len(array2)):
                if array1[i] == array2[j]:
                    isElement = True
                    break

            if not isElement:
                print(array1[i], end=" ")

    def __printDifferentElements_v2(self, array1, array2):
        print("alternative version")

        i = 0
        while i < len(array1):
            for j in range(len(array2)):
                if array1[i] == array2[j]:
                    del array1[i]
                    i = i - 1
                    break
            i = i+ 1

        for i in range(len(array1)):
            print(array1[i], end=" ")

    def printDifferentElements(self, array1, array2, altVersion = False):
        print("The elements of A that are not in B are ")

        if not altVersion:
            self.__printDifferentElements_v1(array1, array2)
        else:
            self.__printDifferentElements_v2(array1, array2)

